types/stake.go

```diff
@@ -58,8 +58,13 @@ type ValidatorSet interface {
 	IterateValidatorsBonded(Context,
 		func(index int64, validator Validator) (stop bool))
 
-	Validator(Context, Address) Validator     // get a particular validator by owner address
-	TotalPower(Context) Rat                   // total power of the validator set
+	Validator(Context, Address) Validator // get a particular validator by owner address
+	TotalPower(Context) Rat               // total power of the validator set
+}
+
+// ValidatorSet that can slash and revoke (affect the validator set)
+type SlashValidatorSet interface {
```

Comment:
Interfaces are clean, but I don't think they accomplish the object-capability guarantees we want; anyone receiving a `ValidatorSet` can cast it to a `SlashValidatorSet` if the underlying type fulfills the latter interface.

Reply:
yeah just adding everything to validator set for now